dkg-gadget/src/worker.rs

```diff
@@ -1099,6 +1107,34 @@ where
 		}
 	}
 
+	/// This function checks
+	/// 1. if the offchain storage of value X is empty, and
+	/// 2. if the corresponding value of X on is not on chain and
+	/// IF the condition is true this means that the offchain storage was cleared without the key
+	/// being written onchain. This function will write the key again to the offchain storage so
+	/// that its pickedup by the offchain workers and tried again.
+	pub fn ensure_generated_keys_are_posted_onchain(
+		&self,
+		latest_header: &B::Header,
+		session_id: SessionId,
+	) {
+		// nextPublicKey
+		if self.get_next_dkg_pub_key(latest_header).is_none() &&
+			is_offchain_storage_empty(self, AGGREGATED_PUBLIC_KEYS)
```

Comment:
I would also add another 3rd check which is to check if `now - self.next_rounds.start_at > N blocks` where `N=5` for example.